File: solutions/c/grains/1/src/grains.c
#include "grains.h"
#include "ctype.h"
#include "stdio.h"

unsigned long long square(unsigned long long n)
{
	if (n==0 || n > 64) return (0);
	unsigned long long g = 1;
	for (unsigned long long i=2; i<=n; i++){
 		g *= 2;         
	}
	return g;
}

unsigned long long total()
{
	unsigned long long t = 0;
	for (unsigned long long i=1; i<=64; i++){
 		t += square(i);         
	}
	return t;
}

File: solutions/fortran/grains/1/grains.f90

module grains

  implicit none

contains

  double precision function square(n)
    integer :: n, i
    if (n <= 0 .or. n >64) then
      square = -1.d0
      return
    end if
    square = 1.d0
    do i = 2,n 
      square=square*2
    end do
    return
  end function

  double precision function total()
  integer :: i
    total = 0.d0
    do i = 1,64
      total=total + square(i)
    end do
    return
  end function

end module
